### ISSUE TITLE
How to bind random local port for simple port forwarding?

### ISSUE THREAD

**Issue by x1a0** [Eugeny/russh#362]
Hi,

I am trying to use the `channel_open_direct_tcpip` method to open a simple local port forwarding, and got 2 questions:
- Is it possible to get a random available local port?
- In my case, I just want to open the local port, so a database client is able to use it to talk to the database via the tunnel. aka I do not really want/need to handle anything via the russh channel. Is it possible?

Thanks

**Comment by Eugeny** [comment]
You might be confused by the difference in concepts between OpenSSH client and the protocol itself. You don't _need_ to open a local port if you just want to connect to somewhere on the remote side via `direct-tcpip`, you can _just do it_ without listening on the local side first.

The reason why the OpenSSH client does this is because that's the only usable way to do it in a general purpose CLI client

If you just want to connect to a remote database, use `channel_open_direct_tcpip`, turn your channel into a stream via `Channel::into_stream` and pass that to your database client library.

**Comment by dcormier** [comment]
> * Is it possible to get a random available local port?

Yes. Specify port `0` as the port to listen on. Most OS's (Linux, Windows, macOS, at least) will assign a random available port. [Example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=8723a6c77b72b474f7dcf7f64a5a23df):

```rust
use tokio::net::TcpListener;

#[tokio::main]
async fn main()  {
    let listener = TcpListener::bind("127.0.0.1:0").await.unwrap();

    println!("Listening on port {}", listener.local_addr().unwrap().port());
}
```

Output (will vary from run to run):
```text
Listening on port 44923
```

You can then pass that `TcpListener` to [`Server::run_on_socket()`](https://docs.rs/russh/latest/russh/server/trait.Server.html#method.run_on_socket).
